runtime/common/metering.go

```diff
@@ -33,3 +38,41 @@ func NewStringMemoryUsage(length int) MemoryUsage {
 		Amount: uint64(length),
 	}
 }
+
+func NewBigIntMemoryUsage(bytes int) MemoryUsage {
+	return MemoryUsage{
+		Kind:   MemoryKindBigInt,
+		Amount: uint64(bytes),
+	}
+}
+
+func max(a, b int) int {
+	if a > b {
+		return a
+	}
+	return b
+}
+
+const bigIntWordSize = int(unsafe.Sizeof(big.Word(0)))
+
+func BigIntByteLength(v *big.Int) int {
+	// NOTE: big.Int.Bits() actually returns bytes:
+	// []big.Word, where big.Word = uint
+	return len(v.Bits()) * bigIntWordSize
+}
+
+func NewPlusBigIntMemoryUsage(a, b *big.Int) MemoryUsage {
+	return NewBigIntMemoryUsage(
+		max(
+			BigIntByteLength(a),
+			BigIntByteLength(b),
+		) + bigIntWordSize,
```

Comment:
Will be good to have a comment with the calculation logic

Reply:
Yes, I'll update the formulas and add comments once https://github.com/dapperlabs/cadence-private-issues/issues/32 is complete